user-scripts/mycourses_exporter.py

```diff
@@ -69,6 +69,7 @@ def export(self, gradebook):
         # Generete a format string for each row entry
         keys = ['username']
         keys.extend(list(map(lambda a : a.name, gradebook.assignments)))
```

Comment:
I assume that no name means that name is `None`, not an empty string?

Maybe this line could be something like `keys.extend(list(map(lambda a : a.name or ``, gradebook.assignments)))`
For now I'm not sure if having assignment with no name is an error.

It does not show up in assignment list?

Reply:
In this case, the problem was an empty string, not None.  At least the problem was that `fmt` below contained `{}`, which I assume only comes from an empty string.  Then, the problem was when `fmt.format(**student_row))` is done, and then `{}` needs a position-based argument, not a keyword argument.

The name we are talking is also the directory name, so an empty string can't even exist... which is why I think that skipping it can't hurt.

The assignment didn't appear in the the `nbgrader list` command line output, or formgrader, and obviously not on the filesystem.

... we can leave this open and see if it ever happens again, then accept it.  Or if it doesn't come soon, close it but remember it.  What do you think?

Reply:
@Sefriol, what do you think - do this or not?

Reply:
IMO just leave it open for now. If it occurs again, maybe take a closer look if there is a recurring problem that should be solved.